usr/lib/common/new_host.c

```diff
@@ -3380,20 +3380,17 @@ CK_RV SC_DigestEncryptUpdate(STDLL_TokData_t *tokdata,
                              CK_ULONG ulPartLen, CK_BYTE_PTR pEncryptedPart,
                              CK_ULONG_PTR pulEncryptedPartLen)
 {
-    UNUSED(sSession);
-    UNUSED(pPart);
-    UNUSED(ulPartLen);
-    UNUSED(pEncryptedPart);
-    UNUSED(pulEncryptedPartLen);
+    CK_RV rc;
 
-    if (tokdata->initialized == FALSE) {
-        TRACE_ERROR("%s\n", ock_err(ERR_CRYPTOKI_NOT_INITIALIZED));
-        return CKR_CRYPTOKI_NOT_INITIALIZED;
-    }
+    rc = SC_EncryptUpdate(tokdata, sSession, pPart, ulPartLen,
+                          pEncryptedPart, pulEncryptedPartLen);
+    if (rc != CKR_OK || pEncryptedPart == NULL)
```

Comment:
In my opinion, the first check is sufficient. But maybe I'm wrong.

Reply:
A size-query call returns CKR_OK, but did not produce any encryption output. So it must not call DigestUpdate in that case.

Reply:
OK, thanks.